### ISSUE TITLE
Failure on Linux Jobs on VSTS

### ISSUE THREAD

**Issue by mikaelm12** [aspnet/SignalR#2788]
Looks like it can't start Chrome headless with out the `--no-sandbox` flag

````
2018-08-10T22:47:30.2380916Z   Checking for ChromeHeadless at /usr/bin/google-chrome...
2018-08-10T22:47:30.2403701Z   Located ChromeHeadless at /usr/bin/google-chrome.
2018-08-10T22:47:30.2420687Z   Checking for ChromiumHeadless at undefined...
2018-08-10T22:47:30.2436042Z   Unable to locate ChromiumHeadless. Skipping.
2018-08-10T22:47:30.2452499Z   Checking for FirefoxHeadless at firefox...
2018-08-10T22:47:30.2467369Z   Unable to locate FirefoxHeadless. Skipping.
2018-08-10T22:47:31.2537932Z   10 08 2018 22:47:31.250:INFO [karma]: Writing browser console to file: /opt/vsts/work/1/s/artifacts/logs/browserlogs.console.2018-08-10T22-47-31-004Z
2018-08-10T22:47:31.2768183Z   10 08 2018 22:47:31.275:INFO [karma]: Karma v2.0.4 server started at http://0.0.0.0:9876/
2018-08-10T22:47:31.2789273Z   10 08 2018 22:47:31.276:INFO [launcher]: Launching browser ChromeHeadless with unlimited concurrency
2018-08-10T22:47:31.2839481Z   10 08 2018 22:47:31.282:INFO [launcher]: Starting browser ChromeHeadless
2018-08-10T22:47:32.5276176Z   10 08 2018 22:47:32.525:ERROR [launcher]: Cannot start ChromeHeadless
2018-08-10T22:47:32.5299402Z   	[0810/224732.423558:ERROR:zygote_host_impl_linux.cc(89)] Running as root without --no-sandbox is not supported. See https://crbug.com/638180.
2018-08-10T22:47:32.5314950Z   
2018-08-10T22:47:32.5329290Z   10 08 2018 22:47:32.525:ERROR [launcher]: ChromeHeadless stdout: 
2018-08-10T22:47:32.5345303Z   10 08 2018 22:47:32.525:ERROR [launcher]: ChromeHeadless stderr: [0810/224732.423558:ERROR:zygote_host_impl_linux.cc(89)] Running as root without --no-sandbox is not supported. See https://crbug.com/638180.
2018-08-10T22:47:32.5362224Z   
2018-08-10T22:47:32.5377731Z   10 08 2018 22:47:32.528:INFO [launcher]: Trying to start ChromeHeadless again (1/2).
2018-08-10T22:47:32.6377894Z   10 08 2018 22:47:32.636:ERROR [launcher]: Cannot start ChromeHeadless
2018-08-10T22:47:32.6394678Z   	[0810/224732.633055:ERROR:zygote_host_impl_linux.cc(89)] Running as root without --no-sandbox is not supported. See https://crbug.com/638180.
2018-08-10T22:47:32.6410202Z   
2018-08-10T22:47:32.6424073Z   10 08 2018 22:47:32.636:ERROR [launcher]: ChromeHeadless stdout: 
2018-08-10T22:47:32.6441028Z   10 08 2018 22:47:32.636:ERROR [launcher]: ChromeHeadless stderr: [0810/224732.633055:ERROR:zygote_host_impl_linux.cc(89)] Running as root without --no-sandbox is not supported. See https://crbug.com/638180.
2018-08-10T22:47:32.6456919Z   
2018-08-10T22:47:32.6472455Z   10 08 2018 22:47:32.639:INFO [launcher]: Trying to start ChromeHeadless again (2/2).
2018-08-10T22:47:32.7471651Z   10 08 2018 22:47:32.745:ERROR [launcher]: Cannot start ChromeHeadless
2018-08-10T22:47:32.7495791Z   	[0810/224732.742165:ERROR:zygote_host_impl_linux.cc(89)] Running as root without --no-sandbox is not supported. See https://crbug.com/638180.
2018-08-10T22:47:32.7514654Z   
2018-08-10T22:47:32.7530099Z   10 08 2018 22:47:32.745:ERROR [launcher]: ChromeHeadless stdout: 
2018-08-10T22:47:32.7641046Z   10 08 2018 22:47:32.745:ERROR [launcher]: ChromeHeadless stderr: [0810/224732.742165:ERROR:zygote_host_impl_linux.cc(89)] Running as root without --no-sandbox is not supported. See https://crbug.com/638180.
2018-08-10T22:47:32.7660834Z   
2018-08-10T22:47:32.7675555Z   10 08 2018 22:47:32.745:ERROR [launcher]: ChromeHeadless failed 2 times (cannot start). Giving up.
2018-08-10T22:47:32.7691287Z   SUMMARY
2018-08-10T22:47:32.7705516Z   No tests did run in any browsers.Terminating dotnet process
2018-08-10T22:47:32.7731693Z   npm ERR! code ELIFECYCLE
2018-08-10T22:47:32.7745773Z   npm ERR! errno 1
2018-08-10T22:47:32.7763095Z   npm ERR! functionaltests@1.0.0 test:local: `ts-node --project ./scripts/tsconfig.json ./scripts/run-tests.ts "--no-color" "--configuration" "Release"`
2018-08-10T22:47:32.7778122Z   npm ERR! Exit status 1
2018-08-10T22:47:32.7792754Z   npm ERR! 
2018-08-10T22:47:32.7806456Z   npm ERR! Failed at the functionaltests@1.0.0 test:local script.
2018-08-10T22:47:32.7822759Z   npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
2018-08-10T22:47:32.8602864Z   
2018-08-10T22:47:32.8619477Z   npm ERR! A complete log of this run can be found in:
2018-08-10T22:47:32.8638934Z   npm ERR!     /root/.npm/_logs/2018-08-10T22_47_32_759Z-debug.log
2018-08-10T22:47:32.8672753Z   npm ERR! code ELIFECYCLE
2018-08-10T22:47:32.8689357Z   npm ERR! errno 1
2018-08-10T22:47:32.8707627Z   npm ERR! functionaltests@1.0.0 test:inner: `npm run build && dotnet build && npm run test:local -- "--no-color" "--configuration" "Release"`
2018-08-10T22:47:32.8725181Z   npm ERR! Exit status 1
2018-08-10T22:47:32.8742448Z   npm ERR! 
2018-08-10T22:47:32.8755492Z   npm ERR! Failed at the functionaltests@1.0.0 test:inner script.
2018-08-10T22:47:32.8771599Z   npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
2018-08-10T22:47:32.8816595Z   
2018-08-10T22:47:32.8834845Z   npm ERR! A complete log of this run can be found in:
2018-08-10T22:47:32.8856972Z   npm ERR!     /root/.npm/_logs/2018-08-10T22_47_32_869Z-debug.log
2018-08-10T22:47:32.8889430Z /opt/vsts/work/1/s/build/repo.targets(48,5): error MSB3073: The command "npm run test:inner -- --no-color --configuration Release" exited with code 1. [/opt/vsts/work/.dotnet/buildtools/korebuild/3.0.0-alpha1-20180731.2/KoreBuild.proj]
2018-08-10T22:47:32.8950091Z 
2018-08-10T22:47:32.8967455Z Build FAILED.
2018-08-10T22:47:32.8974951Z 
2018-08-10T22:47:32.8993611Z /opt/vsts/work/1/s/build/repo.targets(48,5): error MSB3073: The command "npm run test:inner -- --no-color --configuration Release" exited with code 1. [/opt/vsts/work/.dotnet/buildtools/korebuild/3.0.0-alpha1-20180731.2/KoreBuild.proj]
2018-08-10T22:47:32.9009918Z     0 Warning(s)
2018-08-10T22:47:32.9025023Z     1 Error(s)
2018-08-10T22:47:32.9032286Z 
2018-08-10T22:47:32.9124433Z Time Elapsed 00:06:07.14
2018-08-10T22:47:32.9181632Z [0;31merror  : dotnet failed with exit code 1[0m
2018-08-10T22:47:32.9337188Z ##[error]Bash exited with code '1'.
2018-08-10T22:47:32.9423838Z ##[section]Finishing: Run build.sh
```

**Comment by mikaelm12** [comment]
Looks like we might need to create a custom launcher with the ChromeHeadless base and pass the `--no-sandbox` flag

**Comment by mikaelm12** [comment]
https://github.com/karma-runner/karma-chrome-launcher/issues/158

**Comment by BrennanConroy** [comment]
@anurse Fixed right?

**Comment by anurse** [comment]
Fixed in #2796 (https://github.com/aspnet/SignalR/commit/82cef1a0c6324d6f5f5485e1de152db1be93f32f)
